csharp/src/Ice/Connection.cs

```diff
@@ -1658,4 +1609,158 @@ private enum State
             Closed
         };
     }
+
+    public abstract class IPConnection : Connection
+    {
+        public System.Net.IPEndPoint? LocalAddress
+        {
+            get
+            {
+                // Cache the address to be able to access it after the socket is dispose
+                if (_localAddress == null)
+                {
+                    try
+                    {
+                        _localAddress = Transceiver.Fd()?.LocalEndPoint as System.Net.IPEndPoint;
+                    }
+                    catch
+                    {
+                    }
+                }
+                return _localAddress;
```

Comment:
Perhaps the following instead?
```
try
{
    _localAddress ??= Transceiver.Fd()?.LocalEndPoint as System.Net.IPEndPoint;
}
catch
{
}
return _localAddress;
```

Reply:
Removed the caching as suggested in your other comment